proto/p4/config/p4info.proto

```diff
@@ -21,13 +21,11 @@ import "google/protobuf/any.proto";
 package p4.config;
 
 message P4Info {
-  repeated HeaderField header_fields = 1;
```

Comment:
The ID does not really exist in the IR at all. It's purely a function of the P4 object's name and type. The serializer doesn't make ID decisions; it just follows an algorithm for generating IDs.

(There is one exception - `@id` annotations are, of course, present in the IR.)